pkg/download/protocol_test.go

```diff
@@ -294,7 +294,6 @@ func TestInfo(t *testing.T) {
 
 			var info storage.RevInfo
 			dec := json.NewDecoder(bytes.NewReader(bts))
-			dec.DisallowUnknownFields()
```

Comment:
Should we update like this:
https://github.com/gomods/athens/pull/1799/files#diff-34d858fcb9045702dd09c49922588e96b3200e22e41dfc6c0a3201406489396b

Reply:
We can do that too, but I am not sure whether Athens is interested in those additional fields. If we `DisallowUnknownFields`, we have to update the `RevInfo` every time Go add new fields into it, which is an unnecessary maintenance cost if we don't use those new fields.

Reply:
I think we should extend RevInfo to keep it compatible with go proxy. See:
```
➜  ~ curl https://proxy.golang.org/golang.org/x/mod/@v/v0.7.0.info
{"Version":"v0.7.0","Time":"2022-11-08T21:31:42Z","Origin":{"VCS":"git","URL":"https://go.googlesource.com/mod","Ref":"refs/tags/v0.7.0","Hash":"7c05a442b7c1d1a107879b4a090bb5a38d3774a1"}}% ➜  ~ 
```

Reply:
OK. I think #1799 should fix the current CI issue too. Let me close this one then.

Can you restart a CI build for #1799?

Reply:
Sorry, that change was not complete since it was also trying to do a lot more things. Would it be possible for you to expand or takeover that PR as I have not had the time to finish it?

Reply:
I can take the RevInfo related part to a separate PR and land it first, so we can fix CI and unblock other PR. For rest of Go 1.19 upgrade, I will see how far I can get